src/mscorlib/src/System/Guid.cs

```diff
@@ -921,9 +921,12 @@ public override String ToString()
             return ToString("D",null);
         }
 
-        public override int GetHashCode()
+        [System.Security.SecuritySafeCritical]
+        public unsafe override int GetHashCode()
         {
-            return _a ^ (((int)_b << 16) | (int)(ushort)_c) ^ (((int)_f << 24) | _k);
+            // Simply XOR all the bits of the GUID 32 bits at a time.
+            fixed (int* ptr = &this._a)
```

Comment:
It is an interesting idea, but it is unclear which is better.    You need to set up a microbenchmark of a Dictionary<Guid, T> and see if it actually wins.  If it does, we can change it.